### DISCUSSION TITLE
Can't connect neither SQL Server nor MySQL to Budibase

### DISCUSSION THREAD

**Discussion by Tomoxx** [Budibase/budibase#12758] [Help]
Hello, maybe im really dumb but I can't get neither local database to connect to Budibase, 
MySQL just gives:
Error creating datasource: Unable to connect:
SQL Server gives:
Error creating datasource: Unable to connect: Failed to connect to localhost:1433 - Could not connect (sequence)
Can someone give me an example of where to get all the connection data required exactly?

**Comment by ConorWebb96** [comment]
Hey @Tomoxx,

Are you able to connect to your local databases outside of Budibase through a GUI?

If you have made your database in a docker container use `host.docker.internal` as the host parameter and keep everything else the same.

**Reply by Tomoxx** [reply]
I have been able to connect SQL Server to Power Apps and Microsoft Access, also tested the remote connectivity through a .udl file, I don't understand if im really inputting wrong data in a field that Budibase asks for.

Edit: I can't connect from appsmith either, I don't understand what problem there could be.

**Comment by techthrasher** [comment]
I set up my postgresql server in a docker container and had a devil of a time connecting it...until I made sure budibase and postgresql were on the same network, and then I only had to put the container name as the host parameter, and the docker port (not the host port) in the port field.  Connected right up!
